### ISSUE TITLE
Feature request: allow regex rules in the same way as existing text matches

### ISSUE THREAD

**Issue by finnmerlett** [wayou/vscode-todo-highlight#181]
I like this extension, but I would love to be able to specify regex patterns as highlight criteria, rather than the current text-only. I know you can specify the regex highlight pattern, but you can only do one pattern for one set of formatting, and it replaces all your other keyword rules.

It would make more sense to adapt the `todohighlight.keywords` setting to take rules where there is a `"regex"` key allowed instead of the `"text"` key, if you wanted to select by regex for that rule.

My use case is that I want to use TODOhighlight for multiple other separate formatting changes that require individual regexes.

**Comment by jgclark** [comment]
You're in luck, @finnmerlett, despite the author sadly seeming to have abandoned this. A year ago @vonEdfa included this in a PR over in issue #152, and I'm hoping to learn how to get that released as a full new extension. In the meantime if you follow the instructions in #152 you can apply the changes for yourself, and then use a new `"regex"` setting.

**Comment by finnmerlett** [comment]
Thanks! Will have a look

**Comment by aarowman** [comment]
I was looking for exactly this issue. Since the author stopped updating it, how can we get this extension supported again? There are a lot of open PRs that are just sitting there

**Comment by jgclark** [comment]
A long time back I figured out how to fork and publish an extension, so I have done that. Look for TODO Highlight V2, and you'll find what you want.

**Comment by aarowman** [comment]
Nice, I found it and switched over to use that instead. Thanks!

**Comment by Stanzilla** [comment]
Just came across this and wanted to request that as well, nice to see that there is a fork but would love to see this integrated here as well :)

**Comment by jgclark** [comment]
Indeed. But the reason I forked it was that the extension appeared to be
abandoned.
